plugins/chat/assets/javascripts/discourse/lib/chat-threads-manager.js

```diff
@@ -14,34 +14,38 @@ import { TrackedObject } from "@ember-compat/tracked-built-ins";
 */
 
 export default class ChatThreadsManager {
-  @service chatSubscriptionsManager;
   @service chatTrackingStateManager;
   @service chatChannelsManager;
   @service chatApi;
-  @service chat;
-  @service currentUser;
+
   @tracked _cached = new TrackedObject();
 
   constructor(owner) {
     setOwner(this, owner);
   }
 
+  @cached
+  get threads() {
+    return Object.values(this._cached);
+  }
+
   async find(channelId, threadId, options = { fetchIfNotFound: true }) {
-    const existingThread = this.#findStale(threadId);
+    const existingThread = this.#getFromCache(threadId);
     if (existingThread) {
       return Promise.resolve(existingThread);
     } else if (options.fetchIfNotFound) {
-      return this.#find(channelId, threadId);
+      return this.#fetchFromServer(channelId, threadId);
     } else {
       return Promise.resolve();
     }
   }
 
   async index(channelId) {
     return this.chatChannelsManager.find(channelId).then((channel) => {
+      this.#loadIndex(channelId);
```

Comment:
Is this supposed to be doubled up with the line below?

Reply:
Oh I see you are not even using `.index` anymore, nevermind, we can probably get rid of this whole function then